<launch>

    <param name="robot_description" command="$(find xacro)/xacro --inorder $(find ur_custom_motion)/urdf/ur5.xacro" />

    <arg name="x" default="0.0" />
    <arg name="y" default="0.0" />
    <arg name="z" default="0.5" />

    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="use_sim_time" value="true" />
        <arg name="debug" value="false" />
        <arg name="gui" value="true" />
        <arg name="paused" value="true" />
    </include>

    <node name="spawn_ur5" pkg="gazebo_ros" type="spawn_model" output="screen" args="-urdf -model ur5 -param robot_description -x $(arg x) -y $(arg y) -z $(arg z)" />

    <rosparam command="load" file="$(find ur_custom_motion)/config/ur5_controllers.yaml" />

    <node name="controller_spawner" pkg="controller_manager" type="spawner" args="joint_state_controller eff_joint_traj_controller --timeout 60 " ns="/ur5" respawn="false" output="screen" />

    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="false" output="screen">
        <remap from="/joint_states" to="/ur5/joint_states" />
    </node>

    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />

    <remap from="/joint_states" to="/ur5/joint_states" />

    <include file="$(find ur_custom_moveit_config)/launch/planning_context.launch">
        <arg name="load_robot_description" value="false" />
    </include>

    <include file="$(find ur_custom_moveit_config)/launch/move_group.launch">
        <arg name="publish_monitored_planning_scene" value="true" />
    </include>

    <include file="$(find ur_custom_moveit_config)/launch/moveit_rviz.launch">
        <arg name="rviz_config" value="$(find ur_custom_moveit_config)/launch/moveit.rviz" />
    </include>

    <node name="motion_generator" pkg="ur_custom_motion" type="custom_motion_generator.py" />

</launch>